admin login issues

@claude

can you please fix these render deploy issues in api:

2025-10-19T18:36:19.699791563Z Scope: all 3 workspace projects
2025-10-19T18:36:20.422762192Z 
2025-10-19T18:36:20.422787793Z devDependencies: skipped because NODE_ENV is set to production
2025-10-19T18:36:20.422791273Z 
2025-10-19T18:36:20.44074787Z Done in 1.2s
2025-10-19T18:36:20.8492453Z 
2025-10-19T18:36:20.849277611Z > mana-meeples-singles-market@ build:render /opt/render/project/src
2025-10-19T18:36:20.849284011Z > pnpm --filter ./apps/web run build && pnpm --filter ./apps/api run build && node scripts/copy-dist-to-api.js
2025-10-19T18:36:20.849287721Z 
2025-10-19T18:36:21.275944432Z 
2025-10-19T18:36:21.275986073Z > web@0.1.0 build /opt/render/project/src/apps/web
2025-10-19T18:36:21.275990543Z > vite build
2025-10-19T18:36:21.275993373Z 
2025-10-19T18:36:21.499968793Z vite v5.4.20 building for production...
2025-10-19T18:36:21.801244311Z transforming...
2025-10-19T18:36:25.219645246Z ✓ 1621 modules transformed.
2025-10-19T18:36:25.371244137Z rendering chunks...
2025-10-19T18:36:25.471478133Z computing gzip size...
2025-10-19T18:36:25.478815176Z dist/index.html                             0.92 kB │ gzip:  0.50 kB
2025-10-19T18:36:25.478903978Z dist/assets/index-CUHBYSVB.css             44.78 kB │ gzip:  7.78 kB
2025-10-19T18:36:25.47902047Z dist/assets/requestThrottler-CPgOJH9g.js    1.29 kB │ gzip:  0.55 kB
2025-10-19T18:36:25.47903423Z dist/assets/AdminLogin-DJ0tNJC7.js          4.52 kB │ gzip:  1.87 kB
2025-10-19T18:36:25.479122682Z dist/assets/VirtualCardGrid-DIViNZWO.js     4.76 kB │ gzip:  2.00 kB
2025-10-19T18:36:25.479129302Z dist/assets/CurrencySelector-XHOBpCKZ.js    5.62 kB │ gzip:  1.99 kB
2025-10-19T18:36:25.479172123Z dist/assets/AdminDashboard-kQ0YtLr-.js     49.91 kB │ gzip: 10.81 kB
2025-10-19T18:36:25.479249664Z dist/assets/api-cK-G5ABu.js                55.89 kB │ gzip: 13.35 kB
2025-10-19T18:36:25.479367156Z dist/assets/TCGShop-D8mBZ0gs.js            82.68 kB │ gzip: 21.72 kB
2025-10-19T18:36:25.479415087Z dist/assets/index-BkRY_4FP.js             169.41 kB │ gzip: 55.43 kB
2025-10-19T18:36:25.479665202Z ✓ built in 3.96s
2025-10-19T18:36:25.98958212Z 
2025-10-19T18:36:25.9896091Z > @mana-meeples/api@ build /opt/render/project/src/apps/api
2025-10-19T18:36:25.98961257Z > tsc -p tsconfig.json
2025-10-19T18:36:25.98961467Z 
2025-10-19T18:36:29.352674148Z src/routes/auth.ts(98,3): error TS2322: Type '(chunk?: any, encoding?: BufferEncoding | undefined) => Response<any, Record<string, any>>' is not assignable to type '{ (cb?: (() => void) | undefined): Response<any, Record<string, any>>; (chunk: any, cb?: (() => void) | undefined): Response<any, Record<string, any>>; (chunk: any, encoding: BufferEncoding, cb?: (() => void) | undefined): Response<...>; }'.
2025-10-19T18:36:29.352707109Z   Types of parameters 'encoding' and 'cb' are incompatible.
2025-10-19T18:36:29.352710919Z     Type '(() => void) | undefined' is not assignable to type 'BufferEncoding | undefined'.
2025-10-19T18:36:29.352714379Z       Type '() => void' is not assignable to type 'BufferEncoding | undefined'.
2025-10-19T18:36:29.352974784Z src/routes/auth.ts(102,39): error TS2769: No overload matches this call.
2025-10-19T18:36:29.352980514Z   Overload 1 of 3, '(chunk: any, cb?: (() => void) | undefined): Response<any, Record<string, any>>', gave the following error.
2025-10-19T18:36:29.352982954Z     Argument of type 'BufferEncoding | undefined' is not assignable to parameter of type '(() => void) | undefined'.
2025-10-19T18:36:29.352985984Z       Type 'string' is not assignable to type '() => void'.
2025-10-19T18:36:29.352989194Z   Overload 2 of 3, '(chunk: any, encoding: BufferEncoding, cb?: (() => void) | undefined): Response<any, Record<string, any>>', gave the following error.
2025-10-19T18:36:29.352992704Z     Argument of type 'BufferEncoding | undefined' is not assignable to parameter of type 'BufferEncoding'.
2025-10-19T18:36:29.352994984Z       Type 'undefined' is not assignable to type 'BufferEncoding'.
2025-10-19T18:36:29.3867968Z /opt/render/project/src/apps/api:
2025-10-19T18:36:29.38683299Z  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @mana-meeples/api@ build: `tsc -p tsconfig.json`
2025-10-19T18:36:29.38683637Z Exit status 2
2025-10-19T18:36:29.41644271Z  ELIFECYCLE  Command failed with exit code 2.
2025-10-19T18:36:29.449450911Z ==> Build failed 😞
2025-10-19T18:36:29.449482182Z ==> Common ways to troubleshoot your deploy: https://render.com/docs/troubleshooting-deploys

--- a/apps/api/src/routes/auth.ts
+++ b/apps/api/src/routes/auth.ts
@@ -95,11 +95,26 @@ const debugResponse = (req: Request, res: Response, next: express.NextFunction) 
 
   // Override res.end to catch any direct response endings
   const originalEnd = res.end.bind(res);
-  res.end = function(chunk?: any, encoding?: BufferEncoding) {
+  res.end = function(chunk?: any, encoding?: BufferEncoding, cb?: () => void) {
     console.log(`🔍 END: res.end called with chunk:`, chunk);
     logResponseState("before end()");
 
-    const result = originalEnd(chunk, encoding);
+    // Handle the different overloads of res.end()
+    let result;
+    if (typeof encoding === 'function') {
+      // res.end(chunk, callback)
+      result = originalEnd(chunk, encoding);
+    } else if (encoding !== undefined && cb !== undefined) {
+      // res.end(chunk, encoding, callback)
+      result = originalEnd(chunk, encoding, cb);
+    } else if (encoding !== undefined) {
+      // res.end(chunk, encoding)
+      result = originalEnd(chunk, encoding);
+    } else {
+      // res.end(chunk)
+      result = originalEnd(chunk);
+    }
+
     console.log(`🔍 END: res.end execution completed`);
     logResponseState("after end()");
 

